x/ccv/provider/keeper/key_assignment.go

```diff
@@ -40,7 +44,10 @@ func (k Keeper) SetValidatorConsumerPubKey(
 	store := ctx.KVStore(k.storeKey)
 	bz, err := consumerKey.Marshal()
 	if err != nil {
-		panic(err)
+		// An error here would indicate something is very wrong,
+		// the consumer key is obtained from GetValidatorConsumerPubKey, called from
+		// TODO: not sure that consumerKey is actually validated by all possible code paths
```

Comment:
Me neither. @danwt could you please verify that.

Reply:
For InitGenesis, it seems consumerKey is validated properly once #646 is merged. The only other calling function is `AssignConsumerKey`. 

If the consumer key from the [msg server method](https://github.com/cosmos/interchain-security/blob/ba19b5818e249fdc9f4c31a500a9625bd928859f/x/ccv/provider/keeper/msg_server.go#L29) is validated properly, I believe this TODO can be resolved